### DISCUSSION TITLE
Adding table to workspace model - long saving times

### DISCUSSION THREAD

**Discussion by 10xape** [TabularEditor/TabularEditor3#1412] [Data Modeling]
I'm experiencing an unusually long save time in TE3 after adding an empty Measures table to my workspace model. The save operation has been running for over 20 minutes without completing. For context, this is a large model (over 60GB) with multiple tables. Is this extended save duration normal, even for an empty table addition?
![image](https://github.com/user-attachments/assets/5c1c6a95-de5a-4ca6-9246-b25d8aa4c023)
Thank you

Matt

**Comment by 10xape** [comment]
The saving actually failed, not sure what to do from here... 
![image](https://github.com/user-attachments/assets/70f616c6-c865-436b-b5df-4adff69b64ab)

**Reply by otykier** [reply]
Normally it shouldn't take more than a few seconds to save changes to a model. If this issue persists, you should reach out to Microsoft support and inform them that making changes to the model through the XMLA endpoint takes unusually long time. To avoid confusion, you might want to test that the same thing happens if you add a measure to the model using a Microsoft 1st party tool, such as Visual Studio or SQL Server Management Studio (SSMS).

Regarding the error message, please make sure that the empty table you added has a valid M query in its partition. Since I assume this is just going to be a table without any actual data, you can just enter `0` for the M query, like so:

![image](https://github.com/user-attachments/assets/d69bbdde-d379-4d75-8708-f5fdb20878d1)

**Comment by edhans** [comment]
Make sure the model has "large semantic model" turned on in the service. It will take longer to save if this setting is off. However, I've never seen it take more than 2-3 minutes to save with this setting off. The model would need to be well over 200GB or so to even think about taking as long as you are describing.

**Reply by 10xape** [reply]
ok thanks, I have restarted a few times and it finally saved, I iwll check the table with 0
